pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go

```diff
@@ -1621,6 +1959,31 @@ func TestSingleConstraint(t *testing.T) {
 				"node-y": framework.Success,
 			},
 		},
+		{
+			name: "NodeAffinityPolicy honored with labelSelectors",
+			pod: st.MakePod().Name("p").Label("foo", "").
+				NodeSelector(map[string]string{"foo": ""}).
+				SpreadConstraint(2, "zone", v1.DoNotSchedule, fooSelector, nil, nil, nil).
+				Obj(),
+			nodes: []*v1.Node{
+				st.MakeNode().Name("node-a").Label("node", "node-a").Label("foo", "").Obj(),
+				st.MakeNode().Name("node-b").Label("node", "node-b").Label("foo", "").Obj(),
+				st.MakeNode().Name("node-x").Label("node", "node-x").Label("foo", "").Obj(),
+				st.MakeNode().Name("node-y").Label("node", "node-y").Obj(),
+			},
+			existingPods: []*v1.Pod{
+				st.MakePod().Name("p-b1").Node("node-a").Label("foo", "").Obj(),
+				st.MakePod().Name("p-b2").Node("node-b").Label("foo", "").Obj(),
+				st.MakePod().Name("p-y1").Node("node-y").Label("foo", "").Obj(),
+			},
+			enableMinDomains: true,
+			wantStatusCode: map[string]framework.Code{
+				"node-a": framework.Success,
+				"node-b": framework.Success,
+				"node-x": framework.Success,
+				"node-y": framework.Success, // in real case, it's Unschedulable
```

Comment:
If we can't exclude the filtered out nodes in Filter phase, this test seems like a trick. So I created the pr in https://github.com/kubernetes/kubernetes/pull/108866 cc @alculquicondor

Reply:
In theory, someone could disable the NodeAffinity plugin and still run Spreading. So we are actually testing a possible scenario and this should be the behavior.

Reply:
Commented in #108866.

Although it's tricky, it's still possible as Aldo mentioned above. Is it a blocker on your impl.?

Reply:
No blocker. Let's  cover this scenario mentioned above.